internal/sim/api.go

```diff
@@ -359,13 +360,15 @@ func (s *Simulator) Run(duration time.Duration) Results {
 	for i := 0; i < n; i++ {
 		go func() {
 			defer done.Done()
+			r := s.newRegistrar()
```

Comment:
As we discussed offline, let's bake registrar into the Simulator. And declare that the Simulator isn't thread-safe.

Fine to do in a separate change.

Reply:
Will do in a future PR!